crates/host/src/wasmbus/providers/messaging_nats.rs

```diff
@@ -126,79 +246,22 @@ impl wasmcloud_provider_sdk::Provider for Provider {
                 nats.queue_subscribe(async_nats::Subject::from(subject), queue.into())
                     .await
             } else {
-                nats.subscribe(async_nats::Subject::from(sub)).await
+                nats.subscribe(sub).await
             }
             .context("failed to subscribe")?;
-            let target_id = Arc::clone(&target_id);
+            let components = Arc::clone(&self.components);
             let lattice_id = Arc::clone(&self.lattice_id);
             let host_id = Arc::clone(&self.host_id);
-            let components = Arc::clone(&self.components);
+            let target_id = Arc::clone(&target_id);
             tasks.spawn(async move {
                 while let Some(msg) = sub.next().await {
-                    let target_id = Arc::clone(&target_id);
-                    let lattice_id = Arc::clone(&lattice_id);
```

Comment:
While we're at it, we should be removing the subscription here too. If we want to do that in a followup because it's not a part of this PR that's fine, but it might just be the same code

Reply:
Already done